### PR TITLE
TH2-3139. Added check for timestamp of message when added to batch.

### PR DESCRIPTION
**Pull request by MrJonniK**

### CODE REVIEW

File: cradle-core/src/main/java/com/exactpro/cradle/messages/StoredMessageBatch.java [hunk 1]
```diff
@@ -166,24 +162,32 @@ public StoredMessage addMessage(MessageToStore message) throws CradleStorageExce
...
+								+ "' but in your message it is '" + message.getTimestamp() + "'");
 		}
-		
+
+		return addMessageInternal(message);
+	}
+
+	protected StoredMessage addMessageInternal(MessageToStore message)
+	{
+		long messageIndex = message.getIndex() >= 0 ? message.getIndex() : getLastMessage().getIndex()+1;
```
Comment (Vladimir-Panarin-Exactpro) [inline cradle-core/src/main/java/com/exactpro/cradle/messages/StoredMessageBatch.java:182]: `getLastMessage()` can return `null` here.

**Review by Vladimir-Panarin-Exactpro — CHANGES_REQUESTED** [review]

File: cradle-core/src/main/java/com/exactpro/cradle/messages/StoredMessageBatch.java (hunk 1 above)
Comment (MrJonniK) [reply cradle-core/src/main/java/com/exactpro/cradle/messages/StoredMessageBatch.java:182]: I think it is impossible. There are two cases of calling the addMessageInternal() method:
1. Calling from addMessage() method. All needed fields of first message are already checked in addMessage(). When next message hasn't index, first message is there already in messages collection.
2. Calling from DetailedMessageBatchEntity. All messages from db have index.

Comment (MrJonniK) [reply cradle-core/src/main/java/com/exactpro/cradle/messages/StoredMessageBatch.java:182]: But for possible future use, I may add checking for the last message and throwing an exception

File: cradle-core/src/main/java/com/exactpro/cradle/messages/StoredMessageBatch.java [hunk 2]
```diff
@@ -166,24 +162,32 @@ public StoredMessage addMessage(MessageToStore message) throws CradleStorageExce
...
-			else
-				messageIndex = lastMsg.getIndex()+1;
+			if (lastMsg.getTimestamp().isAfter(message.getTimestamp()))
+				throw new CradleStorageException(
+						"Message timestamp should be greater than last message timestamp in batch '" + lastMsg.getTimestamp()
+								+ "' but in your message it is '" + message.getTimestamp() + "'");
 		}
-		
+
+		return addMessageInternal(message);
```
[the remark this answers is not in the document]
Comment (MrJonniK) [reply cradle-core/src/main/java/com/exactpro/cradle/messages/StoredMessageBatch.java:177]: Thanks, It's a good idea

File: cradle-core/src/main/java/com/exactpro/cradle/messages/StoredMessageBatch.java (hunk 1 above)
Comment (MrJonniK) [reply cradle-core/src/main/java/com/exactpro/cradle/messages/StoredMessageBatch.java:182]: I decided not to add the sequence check of the first message in the `addMessageInternal()` method

**Review by Vladimir-Panarin-Exactpro — APPROVED** [review]

**Pull request merged**